go-packages/meep-metrics/gis.go

```diff
@@ -0,0 +1,346 @@
+	metric.Rsrq = int32(rsrq)
+
+	if val, ok = values[UeMetDistance]; !ok {
+		val = ""
+	}
+	metric.Distance = StrToFloat64(val.(string))
+
+	return metric, nil
+}
+
+// GetRedisMetric - Generic metric getter
+func (ms *MetricStore) getGisCacheRedisMetric(metric string, tagStr string) (values []map[string]interface{}, err error) {
+
+        if ms.name == "" {
+                err := errors.New("Store name not specified")
+                return values, err
+        }
+        if ms.redisClient == nil {
+                err = errors.New("Redis metrics DB disabled")
+                return values, err
+        }
+
+	// Get latest metrics
+	//key := gc.baseKey + metric + ":" + tagStr
+	key := ms.baseKeyRef + "gis-cache:" + metric + ":" + tagStr
+
+	err = ms.redisClient.ForEachEntry(key, ms.getMetricsEntryHandler, &values)
+	if err != nil {
+		log.Error("Failed to get entries: ", err)
+		return nil, err
+	}
+	return values, nil
+}
+/*
+func (gc *GisCache) getMetricsEntryHandler(key string, fields map[string]string, userData interface{}) error {
+	// Retrieve field values
+	values := make(map[string]interface{})
+	for k, v := range fields {
+		values[k] = v
+	}
+	// Append values list to data
+	data := userData.(*[]map[string]interface{})
+	*data = append(*data, values)
+
+	return nil
+}
+*/
+func (ms *MetricStore) TakeUeMetricSnapshot() {
+	// start = time.Now()
+
+	// Get all cached metrics
+	valuesArray, err := ms.getGisCacheRedisMetric(UeMetName, "*")
+	if err != nil {
+		log.Error("Failed to retrieve metrics with error: ", err.Error())
+		return
+	}
+
+	// Prepare ue metrics list
+	metricSignalList := make([]Metric, len(valuesArray))
+	metricDistanceList := make([]Metric, len(valuesArray))
+	for index, values := range valuesArray {
+		// Format network metric
+		nm, err := ms.formatCachedUeMetric(values)
+		if err != nil {
+			continue
+		}
+
+		// Add metric to list
+		metricSignal := &metricSignalList[index]
+		metricSignal.Name = UeMetNameInflux
+		metricSignal.Tags = map[string]string{UeMetSrc: nm.Src, UeMetSrcType: nm.SrcType, UeMetDest: nm.Dest, UeMetDestType: nm.DestType, UeMetMeasType: UeMetMeasTypeSignal}
+		metricSignal.Fields = map[string]interface{}{
+			UeMetRssi: nm.Rssi,
+			UeMetRsrp: nm.Rsrp,
+			UeMetRsrq: nm.Rsrq,
+		}
+		metricDistance := &metricDistanceList[index]
+		metricDistance.Name = UeMetNameInflux
+		metricDistance.Tags = map[string]string{UeMetSrc: nm.Src, UeMetSrcType: nm.SrcType, UeMetDest: nm.Dest, UeMetDestType: nm.DestType, UeMetMeasType: UeMetMeasTypeDistance}
+		metricDistance.Fields = map[string]interface{}{
+			UeMetDistance: nm.Distance,
+		}
+
+	}
+
+	// Store metrics in influx
+	err = ms.SetInfluxMetric(metricSignalList)
+	if err != nil {
+		log.Error("Fail to write influx metrics with error: ", err.Error())
+	}
+	// Store metrics in influx
+	err = ms.SetInfluxMetric(metricDistanceList)
+	if err != nil {
+		log.Error("Fail to write influx metrics with error: ", err.Error())
+	}
```

Comment:
Why split them into 2 calls here? We should make a single call to influxdb with the full list of metrics to store.

Reply:
reduced to one call